pkg/csi/service/vanilla/controller.go

```diff
@@ -1806,26 +2050,36 @@ func (c *controller) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequ
 		if common.IsFileVolumeRequest(ctx, volumeCapabilities) {
 			// Error out if TopologyRequirement is provided during file volume provisioning
 			// as this is not supported yet.
-			if req.GetAccessibilityRequirements() != nil {
-				return nil, csifault.CSIInvalidArgumentFault, logger.LogNewErrorCode(log, codes.InvalidArgument,
-					"volume topology feature for file volumes is not supported.")
+			if !commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx, common.TopologyAwareFileVolume) {
+				if req.GetAccessibilityRequirements() != nil {
+					return nil, csifault.CSIInvalidArgumentFault, logger.LogNewErrorCode(log, codes.InvalidArgument,
+						"volume topology feature for file volumes is not supported.")
+				}
 			}
 			volumeType = prometheus.PrometheusFileVolumeType
-			if multivCenterCSITopologyEnabled {
-				isvSANFileServicesSupported, err := c.managers.VcenterManager.IsvSANFileServicesSupported(ctx,
-					c.managers.CnsConfig.Global.VCenterIP)
-				if err != nil {
-					return nil, csifault.CSIInternalFault, logger.LogNewErrorCodef(log, codes.Internal,
-						"failed to verify if vSAN file services is supported or not. Error:%+v", err)
+			if multivCenterCSITopologyEnabled && len(c.managers.VcenterConfigs) > 1 {
+				isvSANFileServicesDisabledInAllVCs := true
+				for _, vcconfig := range c.managers.VcenterConfigs {
+					isvSANFileServicesSupported, err := c.managers.VcenterManager.IsvSANFileServicesSupported(ctx,
+						vcconfig.Host)
+					if err != nil {
+						return nil, csifault.CSIInternalFault, logger.LogNewErrorf(log,
+							"failed to verify if vSAN file services is supported or not for vCenter: %s. "+
+								"Error:%+v", vcconfig.Host, err)
+					}
+					if isvSANFileServicesSupported {
+						isvSANFileServicesDisabledInAllVCs = false
```

Comment:
Here we are basically ensuring that at least 1 VC has FS enabled. This is in contrast to the above logic in Init where all VCs must have FS enabled. We should agree on 1 logic in both the cases.

Reply:
In the Init(), we just have the vc version check and not the FS enablement verification. So we can keep this logic, I think